Code/Mantid/MantidPlot/src/ApplicationWindow.cpp

```diff
@@ -8647,10 +8817,14 @@ void ApplicationWindow::undo()
 
   QApplication::setOverrideCursor(QCursor(Qt::WaitCursor));
 
-  if (qobject_cast<Note*>(w))
-    dynamic_cast<Note*>(w)->editor()->undo();
-  else if (qobject_cast<Matrix*>(w)){
-    QUndoStack *stack = (dynamic_cast<Matrix*>(w))->undoStack();
+  auto note = dynamic_cast<Note*>(w);
+  auto matrix = dynamic_cast<Matrix*>(w);
+
+  if(note)
+    note->editor()->undo();
```

Comment:
Maybe `if(note && note->editor())` for better safety? There are a couple of similar points some 300 lines down...